node/impl/full.go

```diff
@@ -117,4 +123,81 @@ func (n *FullNodeAPI) NodeStatus(ctx context.Context, inclChainStatus bool) (sta
 	return status, nil
 }
 
+// This method is exclusively used for Non Interactive Authentication based on SP Worker Address
+func (n *FullNodeAPI) FilIdSp(ctx context.Context, maddr address.Address) (string, error) {
+	FIL_AUTHHDR := "FIL-SPID-V0"
+
+	head, err := n.ChainHead(ctx)
+	if err != nil {
+		return "", xerrors.Errorf("failed to get chain head: %w", err)
+	}
+
+	beacon, err := n.StateGetBeaconEntry(ctx, head.Height())
+	if err != nil {
+		return "", xerrors.Errorf("failed to get DRAND beacon: %w", err)
+	}
+
+	finality := head.Height() - miner.ChainFinality
+	ftipset, err := n.ChainGetTipSetByHeight(ctx, finality, head.Key())
+	if err != nil {
+		return "", xerrors.Errorf("failed to get finality tipset: %w", err)
+	}
+
+	var msg []byte
+
+	// Append prefix to avoid making a viable message
+	// FIXME: this should be replaced by SPIDV1 combined with proper domain-separation when it arrives
+	msg = append(msg, []byte("   ")...)
+	// Append beacon
+	msg = append(msg, beacon.Data...)
+
+	// Get miner info for last finality
+	minfo, err := n.StateMinerInfo(ctx, maddr, ftipset.Key())
+	if err != nil {
+		return "", xerrors.Errorf("failed to get miner info: %w", err)
+	}
+
+	worker := minfo.Worker
+	var sig *crypto.Signature
+
+	sig, err = n.WalletSign(ctx, worker, msg)
+	if err != nil {
+		return "", xerrors.Errorf("failed to get sign message with worker address key: %w", err)
+	}
+
+	return fmt.Sprintf("%s %d;%s;%s", FIL_AUTHHDR, head.Height(), maddr, string(append([]byte{byte(sig.Type)}, sig.Data...))), nil
```

Comment:
Couple things here:
- the sig needs to be b64 encoded ( it will be used in an HTTP header as-printed )
- as I mentioned earlier - the sigtype is not part of the authid
```suggestion
	return fmt.Sprintf("%s %d;%s;%s", FIL_AUTHHDR, head.Height(), maddr, B64EncodeData(sig.Data))), nil
```